src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -341,6 +341,128 @@ public BitArray Not()
             return this;
         }
 
+        /*=========================================================================
+        ** Shift all the bit values to right on count bits. When count less zero it
+        ** shift all the bit values to left. New bits are be set to zero. The 
+        ** current instance is updated and returned.
+        =========================================================================*/
```

Comment:
Shift operations on `int` and `uint` just give five bits of the count. In fact they make next operations `x shift (count & 0b11111)`.

`BitArray` have any length so we can not give some bits of count.

Left shift is like multiplication by 2^n. `(x << count) == (x * 2 ^ count)`
Right shift is like division by 2^n. `(x >> count) == (x / 2 ^ count)`
And multiplication and division can be defined if count less than 0 because 
`(x * 2 ^ count) == (x / 2 ^ -count)`.